keps/sig-node/129447-log-rotate-on-disk-pressure/README.md

```diff
@@ -0,0 +1,451 @@
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+
+### Risks and Mitigations
+
+No identified risk.
+
+## Design Details
+
+Define 2 new flags `logRotateDiskCheckInterval`, `logRotateDiskPressureThreshold` in kubelet config.
+
+- `logRotateDiskCheckInterval` is the time interval within which the ContainerLogManager will check Disk usage on the kubelet host.
+- `logRotateDiskPressureThreshold` is the threshold of overall Disk usage on the kubelet. If actual Disk usage is equal or more than this threshold, it will rotate logs of all the containers of the kubelet.
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[X] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
+- Add detailed unit tests with 100% coverage.
+- `<package>`: `<date>` - `<test coverage>`
+
+##### Integration tests
+- Scenarios will be covered in e2e tests. 
+
+##### e2e tests
+- Set very high value for `containerLogMaxSize` and `containerLogMaxFiles` to disable periodic log rotation.
+- Add test under `kubernetes/test/e2e_node/container_log_rotation_test.go`.
+- Set very low values for  `logRotateDiskCheckInterval` and `logRotateDiskPressureThreshold`. Create a pod with generating heavy logs and expect the container logs to be rotated after `logRotateDiskCheckInterval` and Disk usage not going more than `logRotateDiskPressureThreshold`.
+
+### Graduation Criteria
+
+**Note:** *Not required until targeted at a release.*
+
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [X] Other
+    - Describe the mechanism:
+    - Will enabling / disabling the feature require downtime of the control
+      plane? Yes (kubelet restart)
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? Yes
```

Comment:
will it? why exactly? could you please elaborate?

Reply:
1. Suppose this is a function of what your cloud provider lets you do in terms of tuning the kubelet configuration. 
2. For on prem deployments, this definitely is just a kubelet restart away. So, not much of a downtime/re-provisioning

Reply:
The cloud provider use case is interesting indeed, but from project perspective this looks like the BM use case anyway. I think a good first step is just adding more details on the answer here.